🪟 🎨 Add style for disabled state to `<CheckBox />` + replace `styled-components` + add `small` size +  add storybook

dizel852 opened the pull request:
## What
Resolves https://github.com/airbytehq/airbyte/issues/20246

## How
Since we don't extend styled components, additional work was also done:
- replace `styled-components` with `css-modules`
- add `small` size variation
- add storybook

Figma:
https://www.figma.com/file/gMNRYVJav25pdsoxAyW7Qp/00_01_WEB-APP-LIBRARIES?node-id=2%3A397&t=2cfAe93e6xY7Gk4L-0

<img width="792" alt="image" src="https://user-images.githubusercontent.com/20929344/206809900-0fec519e-ce75-44db-a065-1cc58401dac0.png">

YatsukBogdan1 commented on airbyte-webapp/src/components/ui/CheckBox/CheckBox.tsx:
```diff
@@ -1,49 +1,38 @@
-  vertical-align: top;
-  position: relative;
-`;
+  small?: boolean;
```
I think we should consider adding the type `CheckBoxSize` (or something similar; the same we have for `Button` component) in case we will have more sizes in the future, so it will be easier to just additional sizes, instead of refactoring all the places where this property was used before

dizel852 replied:
Agree, also was thinking about that. But tried to be in sync with Figma... 
Ok, will change it 👍

dizel852 commented on airbyte-webapp/src/components/ui/CheckBox/CheckBox.tsx:
```diff
@@ -1,49 +1,38 @@
+        className
+      )}
+    >
+      <input type="checkbox" aria-checked={checked} {...inputProps} />
```
Good catch! Yep, we indicate the `indeterminate` only on UI, but semantically it remains unchanged. Will fix that 👍 Thanks!

krishnaglick commented on airbyte-webapp/src/components/ui/CheckBox/CheckBox.tsx:
```diff
@@ -1,49 +1,40 @@
-  vertical-align: top;
-  position: relative;
-`;
+  elSize?: CheckBoxSize;
```
Nit:
```suggestion
  checkboxSize?: CheckBoxSize;
```

timroes replied:
Could we potentially go with `size` here just? It feels given this is the checkbox component whose size we control, we should not try to prefix it with any other prefix, but just use `size` plainly. This would help us also getting to a bit more aligned property names across components.

dizel852 replied:
That was the first I've tried 😄 But unfortunately the native `<input />` has its own `size` property.
We have two options:
1. Omit the native `size` prop and use our one for sizing:
`export interface CheckBoxProps extends Omit<InputHTMLAttributes<HTMLInputElement>, "size"> {
  indeterminate?: boolean;
  size?: CheckBoxSize;
}`
3. Use another prop name for sizing(elSize, checkboxSize, etc.)

@timroes IMO the first option doesn't look dangerous 😬 WDYT?

timroes replied:
Given that the `size` attribute on an input type="checkbox" has anyway no effect (since it's only for text, search, tel, url, email, and password), I think we could safely omit it, so we can nicely reuse it.

dizel852 replied:
Great! Will omit it 👍

dizel852 replied:
Did a try to omit the `size` prop, and here is what I've discovered:
On some higher levels, we through the props via `{...props}`
Here is an example:
<img width="1029" alt="image" src="https://user-images.githubusercontent.com/20929344/207401168-ca593742-1dd0-4f23-bf26-a43bb2f6dbc5.png">

And since we try to change the internal `size` prop, it argues that.

Options we have:
1. Make unavailable to pass props like this way - `{...props}`
2. Use `checkboxSize` as a prop name for sizing

@timroes

dizel852 replied:
Don't wanna block the PR because of the prop name, will change it to `checkboxSize`

krishnaglick reviewed (approved):
Tested in storybook, LGTM

matter-q reviewed (changes requested):
Just a minor thing :wink:

dizel852 commented:
Updated the PR:
1. Changed prop name from `elSize` to `checkboxSize` @krishnaglick @timroes 
2. Added `aria-checked="mixed"` state for  indeterminate state @YatsukBogdan1 
3. Removed unneeded styled checkbox from Sign Up page, [Figma](https://www.figma.com/file/qDwWxPe8NqUMJlvDMikXCr/01_04_SIGN_UP?t=M005x0GPTWrEDtxE-0) for verification

krishnaglick reviewed (approved):
LGTM!

matter-q reviewed (approved):
LGTM!
